node/health.go

```diff
@@ -0,0 +1,81 @@
+package node
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	lapi "github.com/filecoin-project/lotus/api"
+	"github.com/libp2p/go-libp2p-core/network"
+)
+
+type HealthHandler struct {
+	healthy bool
+}
+
+func (h *HealthHandler) SetHealthy(healthy bool) {
+	h.healthy = healthy
+}
+
+func (h *HealthHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	if !h.healthy {
```

Comment:
This isn't thread-safe, slightly better version which will make the race-detector not complain:
```suggestion
	healthy int32
}

func (h *HealthHandler) SetHealthy(healthy bool) {
	h := int32(0)
	if healthy {
		h = 1
	}
	atomic.StoreInt32(&h.healthy, h)
}

func (h *HealthHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
	if atomic.LoadInt32(&h.healthy) != 1 {
```